acceptance/tests/catalog/replace_catalog_debugging.rb

```diff
@@ -92,5 +92,33 @@
 
   end
 
+  step "relationship on resource alias" do
+
+    manifest = <<MANIFEST
+
+    package { foo:
+      ensure => installed,
+      alias=>bar,
+    }
+    service{ baz:
+      ensure => running,
+      require => Package[bar]
+    }
+
+MANIFEST
+
+    run_agents_with_new_site_pp(master, manifest, {}, [0,2,4])
```

Comment:
I would have thought the detailed exit codes were enough to analyze if this passed or not, the remaining test to check events would then become unnecessary. Also, optimally if we could run this with noop, an exit code of 0 or 2 would be enough. I bet now a 4 will always occur, as the package doesn't exist.

For the ref for detailed exist codes, look here:

http://docs.puppetlabs.com/references/4.1.0/man/agent.html

Passing --noop into the extra_cli_args is probably the trick.